apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/DoedshendelseService.kt

```diff
@@ -95,51 +97,78 @@ class DoedshendelseService(
 
     private fun haandterNyeBerorte(
         doedshendelserForAvdoed: List<DoedshendelseInternal>,
-        beroerte: List<Person>,
+        beroerte: List<PersonMedRelasjon>,
         avdoed: PersonDTO,
         endringstype: Endringstype,
     ) {
         val eksisterendeBeroerte = doedshendelserForAvdoed.map { it.beroertFnr }
-        val nyeBeroerte = beroerte.map { it.foedselsnummer.value }.filter { !eksisterendeBeroerte.contains(it) }
-        nyeBeroerte.forEach { barn ->
+        val nyeBeroerte =
+            beroerte.map { PersonFnrMedRelasjon(it.person.foedselsnummer.value, it.relasjon) }
+                .filter { !eksisterendeBeroerte.contains(it.fnr) }
+        nyeBeroerte.forEach { person ->
             doedshendelseDao.opprettDoedshendelse(
                 DoedshendelseInternal.nyHendelse(
                     avdoedFnr = avdoed.foedselsnummer.verdi.value,
                     avdoedDoedsdato = avdoed.doedsdato!!.verdi,
-                    beroertFnr = barn,
-                    relasjon = Relasjon.BARN,
+                    beroertFnr = person.fnr,
+                    relasjon = person.relasjon,
                     endringstype = endringstype,
                 ),
             )
         }
     }
 
+    private fun oppdaterDodshendelser(doedshendelserForAvdoed: List<DoedshendelseInternal>) {
+        doedshendelserForAvdoed.forEach {
+            doedshendelseDao.oppdaterDoedshendelse(it)
+        }
+    }
+
     private fun lagreDoedshendelser(
-        beroerte: List<Person>,
+        beroerte: List<PersonMedRelasjon>,
         avdoed: PersonDTO,
         endringstype: Endringstype,
     ) {
-        beroerte.forEach { barn ->
+        beroerte.forEach { person ->
             doedshendelseDao.opprettDoedshendelse(
                 DoedshendelseInternal.nyHendelse(
                     avdoedFnr = avdoed.foedselsnummer.verdi.value,
                     avdoedDoedsdato = avdoed.doedsdato!!.verdi,
-                    beroertFnr = barn.foedselsnummer.value,
-                    relasjon = Relasjon.BARN,
+                    beroertFnr = person.person.foedselsnummer.value,
+                    relasjon = person.relasjon,
                     endringstype = endringstype,
                 ),
             )
         }
     }
 
-    private fun finnBeroerteBarn(avdoed: PersonDTO): List<Person> {
+    private fun finnBeroerteBarn(avdoed: PersonDTO): List<PersonMedRelasjon> {
         val maanedenEtterDoedsfall = avdoed.doedsdato!!.verdi.plusMonths(1).withDayOfMonth(1)
 
         return with(avdoed.avdoedesBarn ?: emptyList()) {
             this.filter { barn -> barn.doedsdato == null }
                 .filter { barn -> barn.under20PaaDato(maanedenEtterDoedsfall) }
+                .map { PersonMedRelasjon(it, Relasjon.BARN) }
         }
     }
+
+    private fun finnBeroerteEpser(avdoed: PersonDTO): List<PersonMedRelasjon> {
```

Comment:
Mulig jeg er helt på bærtur her, men ser du ikke teknisk sett etter _tidligere_ EPS-er? 
EPS står så vidt jeg vet for "Ektefelle, Partner, og Samboer", men her sjekker du om de _har vært_ EPS.